homeassistant/components/zha/__init__.py

```diff
@@ -134,7 +134,9 @@ async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> b
     else:
         _LOGGER.debug("ZHA storage file does not exist or was already removed")
 
-    zha_gateway = ZHAGateway(hass, config, config_entry)
+    # Re-use the gateway object between ZHA reloads
+    if (zha_gateway := zha_data.get(DATA_ZHA_GATEWAY)) is None:
```

Comment:
Reusing the gateway instance requires special care in the gateway to manage the instance state and life cycle. Have we reviewed all the instance attributes of the gateway and documented that all instance attributes and future changes need to take this into consideration?

Reply:
what about this instead? #99694